webapp/channels/src/components/channel_settings_modal/channel_settings_access_rules_tab.tsx

```diff
@@ -160,6 +166,34 @@ function ChannelSettingsAccessRulesTab({
         setAutoSyncMembers((prev) => !prev);
     }, [parentPolicyAutoSync, expression, autoSyncMembers]);
 
+    // Validate that current user satisfies the expression
+    const validateSelfExclusion = useCallback(async (testExpression: string): Promise<boolean> => {
+        if (!testExpression.trim() || !currentUser?.id) {
+            return true; // No expression or no user, skip validation
+        }
+
+        try {
+            // Test if current user matches the expression
+            const result = await actions.searchUsers(testExpression, '', '', 100);
+            if (result.data) {
+                const matchingUserIds = result.data.users.map((u) => u.id);
```

Comment:
No need to iterate it twice, this can be done in a single iteration-

```typescript
const currentUserMatches = result.data.users.some((u) => u.id === currentUser.id);
```